contracts/interfaces/registries/ILicenseRegistryV2.sol

```diff
@@ -0,0 +1,141 @@
+
+/// @title ILicenseRegistry
+/// @notice This contract is responsible for maintaining relationships between IPs and their licenses,
+/// original and derivative IPs, registering License Templates, setting default licenses,
+/// and managing royalty policies and currency tokens.
+/// It serves as a central point for managing the licensing states within the Story Protocol ecosystem.
+interface ILicenseRegistryV2 {
+    /// @notice Emitted when a new license template is registered.
+    event LicenseTemplateRegistered(address indexed licenseTemplate);
+
+    /// @notice Emitted when a new royalty policy is registered.
+    event RoyaltyPolicyRegistered(address indexed royaltyPolicy);
+
+    /// @notice Emitted when a new currency token is registered.
+    event CurrencyTokenRegistered(address indexed token);
+
+    /// @notice Emitted when a minting license configuration is set.
+    event MintingLicenseConfigSet(
+        address indexed ipId,
+        address indexed licenseTemplate,
+        uint256 indexed licenseTermsId,
+        Licensing.MintingLicenseConfig mintingLicenseConfig
+    );
+
+    /// @notice Emitted when a minting license configuration is set for all licenses of an IP.
+    event MintingLicenseConfigSetForAll(address indexed ipId, Licensing.MintingLicenseConfig mintingLicenseConfig);
+
+    /// @notice Emitted when an expiration time is set for an IP.
+    event ExpireTimeSet(address indexed ipId, uint256 expireTime);
+
+    /// @notice Sets the default license terms that are attached to all IPs by default.
+    /// @param newLicenseTemplate The address of the new default license template.
+    /// @param newLicenseTermsId The ID of the new default license terms.
+    function setDefaultLicenseTerms(address newLicenseTemplate, uint256 newLicenseTermsId) external;
+
+    /// @notice Returns the default license terms.
+    function getDefaultLicenseTerms() external view returns (address licenseTemplate, uint256 licenseTermsId);
+
+    /// @notice Registers a new license template in the Story Protocol.
+    /// @param licenseTemplate The address of the license template to register.
+    function registerLicenseTemplate(address licenseTemplate) external;
+
+    /// @notice Checks if a license template is registered.
+    function isRegisteredLicenseTemplate(address licenseTemplate) external view returns (bool);
+
+    /// @notice Registers a new royalty policy in the Story Protocol.
+    /// @param royaltyPolicy The address of the royalty policy to register.
+    function registerRoyaltyPolicy(address royaltyPolicy) external;
+
+    /// @notice Checks if a royalty policy is registered.
+    function isRegisteredRoyaltyPolicy(address royaltyPolicy) external view returns (bool);
+
+    /// @notice Registers a new currency token used for paying license token minting fees and royalties.
+    /// @param token The address of the currency token to register.
+    function registerCurrencyToken(address token) external;
```

Comment:
any permission control on this function?

Reply:
yes, only protocol admin can access the function, the access control defined in the implementation contract.